### DISCUSSION TITLE
What exactly is meant by the "Invalid flac header" message & what can be done?

### DISCUSSION THREAD

**Discussion by DeathStalker77** [nicotine-plus/nicotine-plus#2756] [Q&A]
I haven't found a clear explanation on this.  Does it _really_ make a difference in the file/s?  Is there a way to correct it?

Thanks!

--- DS

**Comment by mathiascode** [comment]
It can mean one of three things:
- The file has the wrong file extension (e.g. an mp3 file someone renamed to flac)
- The file is corrupted
- The audio metadata parser is not parsing the file properly

If you send the files with this issue to my email address on my GitHub profile, I could take a look.

**Comment by DeathStalker77** [comment]
Well I did check them and they would not play with VLC.  So that begs the question, is there any way to check for those issues during the download?  Thanks! :-)

**Reply by mathiascode** [reply]
You could try using something like https://losslessaudiochecker.com/ to bulk-check downloaded files. I would take the results with a grain of salt though, there might be several false positives (I haven't tried the tool myself).

What we could possibly try in Nicotine+ itself is to also run downloaded audio files through our metadata parser, and log any error messages.

**Comment by DeathStalker77** [comment]
Question - JDownloader has it's own internal CRC check.  I have no idea how it works, but could something like that also be added?  Just to help ensure the validity of DLs.  Also, I don't know if it's possible to do, but what about the idea of internally running a check against any FFP or MD5 files when the applicable folder/s finish?  :-)

**Reply by slook** [reply]
That may be possible for you to implement this using a command line triggered in Preferences > Download > Download Events.

Alternatively, if output into the Nicotine+ status log is desired, then it could be implemented in a custom plugin.

**Comment by DeathStalker77** [comment]
Hmm, personally, I'm not that good with CLI  :-(  Hopefully then, someone will be able to design a plug-in if it cannot be effectively integrated into the client :-)
